src/figuresClasses.ts

```diff
@@ -1,19 +1,74 @@
-export interface Figure {
+type Color = 'red' | 'green' | 'blue';
+type Shape = 'triangle' | 'circle' | 'rectangle';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public aSide: number,
+    public bSide: number,
+    public cSide: number,
+  ) {
+    if (aSide <= 0 || bSide <= 0 || cSide <= 0) {
+      throw new Error('The side of triangle is less than or equal to zero');
+    }
```

Comment:
you have the same functionality in each constructor: you checked that all input valued should be greater or equal to 0
let's create separate function for this check, and will call it for each figure